set.mm

```diff
@@ -551128,6 +551128,23 @@ relation with an intersection with the same Cartesian product (cf.
       BEAUQVEUQVEIUPAGUPBUGUKUHEBAVDUIVFABEAULUMUJUN $.
   $}
 
+  $( Existence of an image of a class (see e.g. ~ cnvepimaex ).
+     Weakened version of theorem 3.17 of [Monk1] p. 39.
+     (Contributed by Peter Mazsa, 22-Feb-2023.) $)
+  imaexALTV $p |- ( ( A e. V \/ ( ( A |` B ) e. W /\ B e. X ) ) ->
```

Comment:
@avekens this way?

Reply:
https://github.com/metamath/set.mm/pull/3318/commits/15227b455523577c76b8f9ad59f7a34d24b686c8

Reply:
@avekens [similarly](https://github.com/metamath/set.mm/pull/3318/commits/d109bbd9865b2a8278a3affda12aa1c787194fac)

Reply:
Doesn't this make the theorem _stronger_ if the hypothesis is _weaker_?
In terms of naming convention, maybe this could be named `ecexs` ?

Reply:
> Doesn't this make the theorem _stronger_ if the hypothesis is _weaker_? 

Are you sure? Seems to be logical what you say but I don't know the related math conventions. If you are sure about this, I'll change it something similar to strengthening the theorem by weakening the antecedent, if you are not, then I'll write only about the weakening of the antecedent and be silent about the strength of the theorem:)

>  In terms of naming convention, maybe this could be named `ecexs` ?

I did not find anything about the *s suffix in our Conventions: is it for a stronger version of a theorem?
I think we should make these replacements in the future: ecexg -> ecexgOLD and ecexALTV -> ecex ; this is why I used this *ALTV label.

Reply:
@tirix safe bet: now it's only about weakening of antecedents https://github.com/metamath/set.mm/pull/3318/commits/ac8a55ce5821883075de76fa9feac5cb1af542ee

Reply:
[Math StackExchange has a definition](https://math.stackexchange.com/a/53712) of stronger/weaker statements. In your case your theorem implies `ecexg`, but the converse is not true so your theorem is stronger.

> I think we should make these replacements in the future: ecexg -> ecexgOLD and ecexALTV -> ecex

The theorem `ecexg` is already used 28 times in `set.mm` so my point here was that although it is weaker, I believe the current version of `ecexg` is easier to use. So maybe a possible future renaming should rather be something like `ecexg` -> `ecexgw`, `ecexALTV -> ecex`, keeping the current version in main.

Reply:
> Math StackExchange has a definition

Thank you very much for the reference, now it is all clear.

> The theorem ecexg is already used 28 times in set.mm so my point here was that although it is weaker, I believe the current version of ecexg is easier to use. So maybe a possible future renaming should rather be something like ecexg -> ecexgw, ecexALTV -> ecex, keeping the current version in main.

Here we disagree, and now all theorems are in my Mathbox which are necessary to demonstrate my point of view. 

The most time consuming activity for me in Metamath was to find a way out of the forest of 1. not perfect definitions and 2. too weak theorems. Ad 2, for example, I found ~ uniqsALTV at the origin of the ~ uniqsALTV <- ~ rnresequniqs <- ~ unidmqs <- ~ unidmqseq <- ~ dmqseqim <- ~ dmqseqim2 <- ~ erim2 <- ~ erim sequence, and, believe me, it was not a linear / straightforward discovery. I don't wish anyone to have to make this journey unnecessarily again. If we allow some weaker theorems while we already have the stronger ones, then we are wasting the time of both the original and the future submitters who directly or indirectly depend on this theorem, and that is completely unnecessary.

Yes, I'm lazy and did not submit this to the main part of the set.mm because that would need to make all theorems stronger which depend directly or indirectly from ~ uniqsALTV etc. But if I will be alert and have any opportunity to influence the collective position on this issue when somebody will really badly want to see this item in the main, I will insist that the precondition of moving ~ uniqsALTV out of my Mathbox is making *all* theorems stronger down the line:)

Reply:
of course I'll help in this non-transparent amount of work, either when needed, or, if we preemptively decide that we'll we carry the changes through set.mm